x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/index.test.tsx

```diff
@@ -26,16 +27,167 @@ jest.mock('react-router-dom', () => {
 jest.mock('../../../containers/detection_engine/lists/use_lists_config');
 jest.mock('../../../../common/components/link_to');
 jest.mock('../../../components/user_info');
-jest.mock('../../../containers/detection_engine/rules');
+jest.mock('../../../../common/components/toasters', () => {
+  const actual = jest.requireActual('../../../../common/components/toasters');
+  return {
+    ...actual,
+    errorToToaster: jest.fn(),
+    useStateToaster: jest.fn().mockReturnValue([jest.fn(), jest.fn()]),
+    displaySuccessToast: jest.fn(),
+  };
+});
+
+jest.mock('../../../containers/detection_engine/rules/api', () => ({
+  getPrePackagedRulesStatus: jest.fn().mockResolvedValue({
+    rules_not_installed: 0,
+    rules_installed: 0,
+    rules_not_updated: 0,
+    timelines_not_installed: 0,
+    timelines_installed: 0,
+    timelines_not_updated: 0,
+  }),
+  createPrepackagedRules: jest.fn(),
+}));
+
+jest.mock('../../../../common/lib/kibana', () => {
+  return {
+    useToast: jest.fn(),
+    useHttp: jest.fn(),
+  };
+});
+
+jest.mock('../../../components/value_lists_management_modal', () => {
+  return {
+    ValueListsModal: jest.fn().mockReturnValue(<div />),
+  };
+});
+
+jest.mock('./all', () => {
+  return {
+    AllRules: jest.fn().mockReturnValue(<div />),
+  };
+});
+
+jest.mock('../../../../common/utils/route/spy_routes', () => {
+  return {
+    SpyRoute: jest.fn().mockReturnValue(<div />),
+  };
+});
+
+jest.mock('../../../components/rules/pre_packaged_rules/update_callout', () => {
+  return {
+    UpdatePrePackagedRulesCallOut: jest.fn().mockReturnValue(<div />),
+  };
+});
 
 describe('RulesPage', () => {
   beforeAll(() => {
     (useUserData as jest.Mock).mockReturnValue([{}]);
-    (usePrePackagedRules as jest.Mock).mockReturnValue({});
   });
-  it('renders correctly', () => {
+
+  it('renders AllRules', () => {
     const wrapper = shallow(<RulesPage />);
+    expect(wrapper.find('[data-test-subj="all-rules"]').exists()).toEqual(true);
+  });
+
+  it('renders correct button with correct text - Load Elastic prebuilt rules and timeline templates', async () => {
```

Comment:
could you please refactor all of your tests to use `waitFor` instead of `act`? Whenever you see that `act` error on a test, you should resolve it using `waitFor` with enzyme `mount` and `act` with `renderHook` You can follow the `waitFor` pattern in [edit_connector/index.test.tsx](https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/public/cases/components/edit_connector/index.test.tsx). I'll show you an example on this test:

```
import { waitFor } from '@testing-library/react';

  it('renders correct button with correct text - Load Elastic prebuilt rules and timeline templates', async () => {
    (getPrePackagedRulesStatus as jest.Mock).mockResolvedValue({
      rules_not_installed: 3,
      rules_installed: 0,
      rules_not_updated: 0,
      timelines_not_installed: 3,
      timelines_installed: 0,
      timelines_not_updated: 0,
    });

    const wrapper = mount(
      <TestProviders>
        <RulesPage />
      </TestProviders>
    );
    await waitFor(() => {
      wrapper.update();
      expect(wrapper.find('[data-test-subj="loadPrebuiltRulesBtn"]').exists()).toEqual(true);
      expect(wrapper.find('[data-test-subj="loadPrebuiltRulesBtn"]').last().text()).toEqual(
        'Load Elastic prebuilt rules and timeline templates'
      );
    });
  });
```
```

Reply:
if you get stuck and want to zoom let me know!